### PR TITLE
Added more docs for ref fields

### PR DESCRIPTION
**Pull request by pkulikov**

### CODE REVIEW

File: docs/csharp/language-reference/builtin-types/ref-struct.md [hunk 1]
```diff
@@ -30,15 +28,32 @@ To declare a `ref struct` as `readonly`, combine the `readonly` and `ref` modifi
 
 In .NET, examples of a `ref struct` are <xref:System.Span%601?displayProperty=nameWithType> and <xref:System.ReadOnlySpan%601?displayProperty=nameWithType>.
 
+## `ref` fields
+
+Beginning with C# 11, you can declare a `ref` field in a `ref struct`, as the following example shows:
+
+:::code language="csharp" source="snippets/shared/StructType.cs" id="SnippetRefField":::
+
+A `ref` field may have the `null` value. Use the <xref:System.Runtime.CompilerServices.Unsafe.IsNullRef%60%601?displayProperty=nameWithType> method to determine if a `ref` field is `null`.
```
Comment (pkulikov) [inline docs/csharp/language-reference/builtin-types/ref-struct.md:37]: @BillWagner could you please help me to fix this xref to https://learn.microsoft.com/en-us/dotnet/api/system.runtime.compilerservices.unsafe.isnullref?view=net-6.0

File: docs/csharp/language-reference/builtin-types/ref-struct.md [hunk 2]
```diff
@@ -30,15 +28,32 @@ To declare a `ref struct` as `readonly`, combine the `readonly` and `ref` modifi
 
 In .NET, examples of a `ref struct` are <xref:System.Span%601?displayProperty=nameWithType> and <xref:System.ReadOnlySpan%601?displayProperty=nameWithType>.
 
+## `ref` fields
+
+Beginning with C# 11, you can declare a `ref` field in a `ref struct`, as the following example shows:
+
+:::code language="csharp" source="snippets/shared/StructType.cs" id="SnippetRefField":::
+
+A `ref` field may have the `null` value. Use the <xref:System.Runtime.CompilerServices.Unsafe.IsNullRef%601?displayProperty=nameWithType> method to determine if a `ref` field is `null`.
```
Comment (BillWagner) [inline docs/csharp/language-reference/builtin-types/ref-struct.md:37]: Here's the link I get:

```suggestion
A `ref` field may have the `null` value. Use the <xref:System.Runtime.CompilerServices.Unsafe.IsNullRef%60%601(%60%600@)?displayProperty=nameWithType> method to determine if a `ref` field is `null`.
```

File: docs/csharp/language-reference/builtin-types/ref-struct.md [hunk 3]
```diff
@@ -30,15 +28,32 @@ To declare a `ref struct` as `readonly`, combine the `readonly` and `ref` modifi
...
+
+You can apply the `readonly` modifier to a `ref` field in the following ways:
```
Comment (BillWagner) [inline docs/csharp/language-reference/builtin-types/ref-struct.md:39]: I'd add a sentence here, or at the end of this section, that the `readonly` keyword impacts the token to its right. In other words, `readonly ref` means you can't `ref` reassign. `ref readonly int n` means you can't reassign the value of `n`.

**Review by BillWagner — COMMENTED** [review]
This is great @pkulikov 

I've made a not on the xref, and had one other minor suggestion to consider. Then, I'll give it another look and we can :shipit:

File: docs/csharp/language-reference/builtin-types/ref-struct.md (hunk 2 above)
Comment (pkulikov) [reply docs/csharp/language-reference/builtin-types/ref-struct.md]: Nice, thank you!

File: docs/csharp/language-reference/builtin-types/ref-struct.md (hunk 3 above)
Comment (BillWagner) [reply docs/csharp/language-reference/builtin-types/ref-struct.md:39]: @pkulikov I see your point. I'll approve this, and :shipit: 

I'll keep an eye on any feedback to see if we want refinements later.

**Review by BillWagner — APPROVED** [review]
Thanks again @pkulikov 

Let's :shipit:

**Pull request merged**